tests/tests/test-proxy.ts

```diff
@@ -0,0 +1,80 @@
+import { expect } from "chai";
+import { customWeb3Request } from "../util/providers";
+import { describeDevMoonbeam } from "../util/setup-dev-tests";
+import Keyring from "@polkadot/keyring";
+import { GENESIS_ACCOUNT_PRIVATE_KEY } from "../util/constants";
+
+function delay(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+describeDevMoonbeam("Pallet proxy", (context) => {
+  it("shouldn't accept unknown proxy", async function () {
+    const keyring = new Keyring({ type: "ethereum" });
+    const genesisAccount = await keyring.addFromUri(GENESIS_ACCOUNT_PRIVATE_KEY, null, "ethereum");
+    const alice = keyring.addFromUri(
+      "sample split bamboo west visual approve brain fox arch impact relief smile"
+    );
+
+    // Give alice some money.
+    await context.polkadotApi.tx.balances
+      .transfer(alice.address, 2_000_000_000)
+      .signAndSend(genesisAccount);
+    await context.createBlock();
+
+    // Don't allow proxy.
+
+    // Proxy call
+    const unsub = await context.polkadotApi.tx.proxy
+      .proxy(
+        genesisAccount.address,
+        null,
+        context.polkadotApi.tx.balances.transfer(alice.address, 100)
```

Comment:
who's signing this call?

Reply:
Nobody is signing the inner call. This call to `transfer` is encoded as a parameter of the outer `proxy` call (which is signed by Alice).

Reply:
I see 👍